src/encoded/static/components/visualizations.js

```diff
@@ -244,13 +246,273 @@ export class TestViz extends React.Component {
+            }
+        });
+        // sort the fake facets
+        const keys = Object.keys(fakeFacets);
+        keys.sort((a, b) => (fakeFacets[b] - fakeFacets[a]));
+        const sortedFakeFacets = [];
+        keys.forEach((key) => {
+            sortedFakeFacets[key] = fakeFacets[key];
+        });
+        const chartData = sortedFakeFacets;
+        const chartTitle = this.props.displayTitle;
+        const chartArray = Object.keys(chartData).map(key => chartData[key]);
+        const chartMax = Math.max(...chartArray);
+        // compute left margin
+        let leftMargin = 60;
+        Object.keys(chartData).forEach((d) => {
+            leftMargin = Math.max(d.length * 7, leftMargin);
+        });
+        // add in some extra margin for white space and caret icon
+        leftMargin += 50;
+        this.setState({
+            chartData,
+            chartTitle,
+            chartMax,
+            leftMargin,
+            data,
+        });
+    }
+
+    render() {
+        let errorMessage = false;
+        if (this.state.searchTerm) {
+            errorMessage = true;
+        }
+        return (
+            <div className="bar-chart-container">
+                <div className="bar-chart-header">
+                    <h4>{this.state.chartTitle}</h4>
+                    <button onClick={this.expandTerms}><i className="icon icon-expand" /></button>
+                    <button onClick={this.collapseTerms}><i className="icon icon-compress" /></button>
+                    <div className="chart-typeahead-container">
+                        <div className="chart-typeahead" role="search">
+                            <i className="icon icon-search" />
+                            <div className="searchform">
+                                <input type="search" aria-label="search to filter biosample results" placeholder="Search for a biosample name" value={this.state.unsanitizedSearchTerm} onChange={this.handleSearch} />
+                            </div>
+                            <i className="icon icon-times" aria-label="clear search and see all biosample results" onClick={this.clearSearch} onKeyDown={this.clearSearch} role="button" tabIndex="0" />
+                        </div>
+                    </div>
+                </div>
+                {Object.keys(this.state.chartData).map((d, dIDX) => {
+                    let searchTermMatch = true;
+                    if (this.state.searchTerm) {
+                        searchTermMatch = sanitizedString(d).match(this.state.searchTerm);
+                        if (searchTermMatch) {
+                            errorMessage = false;
+                        }
+                    }
+                    return (
+                        <div
+                            className={`biosample-table table${dIDX} ${searchTermMatch ? 'display-table' : ''}`}
+                            key={`table${dIDX}`}
+                        >
```

Comment:
If you can avoid using a loop index as a key (and I’m not sure if you can here or not) and instead use some data unique to each array element, that could avoid problems if the order of chartData changes.